oxidation/libparsec/crates/libparsec_fs/src/file_operations.rs

```diff
@@ -39,26 +23,53 @@ fn block_read(chunks: &[Chunk], size: u64, start: u64) -> impl Iterator<Item = C
     // Loop over chunks
     chunks
         .get(start_index..stop_index)
-        .unwrap()
+        .expect("Indexes are found using binary search and hence always valid")
         .iter()
         .map(move |chunk| {
             let mut new_chunk = chunk.clone();
             new_chunk.start = max(chunk.start, start);
-            new_chunk.stop = min(chunk.stop, NonZeroU64::new(stop).unwrap());
+            new_chunk.stop = min(
+                chunk.stop,
+                NonZeroU64::new(stop)
+                    .expect("The stop offset can only be 0 if the index range is empty"),
+            );
             new_chunk
         })
 }
 
-pub fn prepare_read(manifest: LocalFileManifest, size: u64, offset: u64) -> Vec<Chunk> {
+pub fn prepare_read(manifest: &LocalFileManifest, size: u64, offset: u64) -> Vec<Chunk> {
```

Comment:
It just that I prefer to use `usize` / `isize` when possible. But if I take it's definition from the doc 

> The size of this primitive is how many bytes it takes to reference any location in memory. 
> For example, on a 32 bit target, this is 4 bytes and on a 64 bit target, this is 8 bytes.

It may play against us if we're running on a 32-bit system

Reply:
So it's a matter of opinion? How do we decide which one to use?

Reply:
We shouldn't use usize or isize if the data is serialized/deserialized and this is the case of Block/Chunk
usize and isize are great for fitting the architecture